lib/diplomat/kv.rb

```diff
@@ -86,18 +77,71 @@ def get(key, options = {}, not_found = :reject, found = :return)
       end
 
       # Wait for first/next value
-      custom_params << use_named_parameter('index', index)
-      if options.nil?
-        options = { timeout: 86_400 }
-      else
-        options[:timeout] = 86_400
-      end
-      @raw = send_get_request(@conn, ["/v1/kv/#{@key}"], options, custom_params)
+      @raw = wait_for_value(index, options)
       @raw = parse_body
       return_value(return_nil_values, transformation)
     end
     # rubocop:enable PerceivedComplexity, LineLength, MethodLength, CyclomaticComplexity
 
+    # Get all keys recursively, potentially blocking for the first or next value
+    # @param key [String] the key
+    # @param options [Hash] the query params
+    # @option options [String] :consistency The read consistency type
+    # @option options [String] :dc Target datacenter
+    # @option options [Boolean] :keys Only return key names.
+    # @option options [Boolean] :decode_values Return consul response with decoded values.
+    # @option options [String] :separator List only up to a given separator.
+    #   Only applies when combined with :keys option.
+    # @option options [Boolean] :nil_values If to return keys/dirs with nil values
+    # @option options [Boolean] :convert_to_hash Take the data returned from consul and build a hash
+    # @option options [Callable] :transformation function to invoke on keys values
+    # @param not_found [Symbol] behaviour if the key doesn't exist;
+    #   :reject with exception, :return degenerate value, or :wait for it to appear
+    # @param found [Symbol] behaviour if the key does exist;
+    #   :reject with exception, :return its current value, or :wait for its next value
+    # @return [List] List of hashes, one hash for each key-value returned
+    # rubocop:disable PerceivedComplexity, MethodLength, LineLength, CyclomaticComplexity
+    def get_all(key, options = {}, not_found = :reject, found = :return)
+      @options = options
+      @options[:recurse] = true
+      return_nil_values = @options && @options[:nil_values]
+      transformation = @options && @options[:transformation] && @options[:transformation].methods.find_index(:call) ? @options[:transformation] : nil
+
+      raw = get_raw(key, options)
+      if raw.status == 404
+        case not_found
+        when :reject
+          raise Diplomat::KeyNotFound, key
+        when :return
+          return @value = ''
```

Comment:
This looks incorrect to me, I would instead return the empty array.

Reply:
@alexdglover would you mind fixing the `@value = ''` so I can merge this?

This is very good otherwise, I we can merge it as soon you fixed this little issue.

Thank you very much for your contribution

Reply:
Ah good catch. Sorry for the delay, will fix that momentarily